packages/maskbook/src/plugins/Wallet/services/phrase.ts

```diff
@@ -112,3 +113,55 @@ export async function deriveWalletFromPhrase(
     throw new Error('Derive too many times.')
 }
 //#endregion
+
+//#region Query derivable wallet addresses
+export async function queryDerivableWalletFromPhrase(
+    mnemonic: string[],
+    passphrase: string,
+    page: number,
+    pageSize = 10,
+    path = HD_PATH_WITHOUT_INDEX_ETHEREUM,
+) {
+    const result = []
+    for (let i = pageSize * (page - 1); i < pageSize * page; i += 1) {
+        const derivedWallet = await wallet.recoverWalletFromMnemonicWords(mnemonic, passphrase, `${path}/${i}`)
+        const balance = await getBalance(derivedWallet.address)
```

Comment:
Why not fetch balance in the front end?

Reply:
Because `getBalance` can only run in the ManifestBackground context.